src/pkg/egress/syslog/https.go

```diff
@@ -32,15 +42,87 @@ func NewHTTPSWriter(
 ) egress.WriteCloser {
 
 	client := httpClient(netConf, tlsConf)
+	if binding.URL.Query().Get("batching") == "true" {
+		return &BatchHTTPSWriter{
+			HTTPSWriter: HTTPSWriter{
+				url:             binding.URL,
+				appID:           binding.AppID,
+				hostname:        binding.Hostname,
+				client:          client,
+				egressMetric:    egressMetric,
+				syslogConverter: c,
+			},
+			msgBatch:     "",
+			batchSize:    BATCHSIZE,
+			sendInterval: time.Second,
+		}
+	} else {
+		return &HTTPSWriter{
+			url:             binding.URL,
+			appID:           binding.AppID,
+			hostname:        binding.Hostname,
+			client:          client,
+			egressMetric:    egressMetric,
+			syslogConverter: c,
+		}
+	}
+}
+
+func (w *BatchHTTPSWriter) sendMsgBatch() error {
+	req := fasthttp.AcquireRequest()
+	req.SetRequestURI(w.url.String())
+	req.Header.SetMethod("POST")
+	req.Header.SetContentType("text/plain")
+	req.SetBodyString(w.msgBatch)
+
+	w.msgBatch = ""
+	w.sendTimer = nil
+
+	resp := fasthttp.AcquireResponse()
+
+	err := w.client.Do(req, resp)
+	if err != nil {
+		return w.sanitizeError(w.url, err)
+	}
+
+	if resp.StatusCode() < 200 || resp.StatusCode() > 299 {
+		return fmt.Errorf("syslog Writer: Post responded with %d status code", resp.StatusCode())
+	}
```

Comment:
Seems like much of this functionality is shared with the non-batch writer, could refactor into a private func?

Reply:
possibly true, will think about it